src/app/components/wrapper-components/part-creation/part-creation.component.ts

```diff
@@ -18,98 +21,193 @@ interface ComparableSubscription {
-
-    public acceptingDisabled: boolean = true;
-    public proposingDisabled: boolean = true;
-    public proposalSent: boolean = false;
-
-    // Game Configuration Values
-    // timeout = 60;
-    public firstPlayer: string = 'CREATOR';
-    public maximalMoveDuration: number = 30;
+    public joinerObs: Observable<IJoiner>;
+    public canEditConfigObs: Observable<boolean>;
+    public canProposeConfigObs: Observable<boolean>;
+    public canReviewConfigObs: Observable<boolean>;
+    public userIsCreatorObs: Observable<boolean>;
+    public userIsChosenPlayerObs: Observable<boolean>;
+    public userIsObserverObs: Observable<boolean>;
+    public userIsModifyingConfigObs: Observable<boolean>;
+    public showCustomTimeObs: Observable<boolean>;
 
     // Subscription
     private candidateSubscription: MGPMap<string, ComparableSubscription> = new MGPMap();
+    private creatorSubscription: () => void = null;
 
-    public opponentFormGroup: FormGroup;
     public configFormGroup: FormGroup;
 
     public constructor(public router: Router,
                        public gameService: GameService,
                        public joinerService: JoinerService,
                        public chatService: ChatService,
                        public userService: UserService,
-                       public formBuilder: FormBuilder) {
+                       public formBuilder: FormBuilder,
+                       public messageDisplayer: MessageDisplayer) {
         display(PartCreationComponent.VERBOSE, 'PartCreationComponent constructed for ' + this.userName);
     }
     public async ngOnInit(): Promise<void> {
         display(PartCreationComponent.VERBOSE, 'PartCreationComponent.ngOnInit for ' + this.userName);
 
-        this.checkEntry();
+        this.checkInputs();
         this.createForms();
-        try {
-            await this.joinerService.joinGame(this.partId, this.userName);
-        } catch (error) {
-            // ABORT JoinerObserving and page construction, OnlineGameWrapperComponent will soon enough redirect
-            return;
+        const gameExists: boolean = await this.joinerService.joinGame(this.partId, this.userName);
+        if (gameExists === false) {
+            // We will be redirected by the GameWrapper
+            return Promise.resolve();
         }
-        this.joinerService.startObserving(this.partId, (iJoinerId: IJoinerId) =>
-            this.onCurrentJoinerUpdate(iJoinerId));
+        this.joinerObs = this.joinerService.observe(this.partId).pipe(map((id: IJoinerId): IJoiner => {
+            return id.doc;
+        }));
+        this.canReviewConfigObs = this.joinerObs.pipe(map((joiner: IJoiner): boolean => {
+            return joiner.partStatus === PartStatus.CONFIG_PROPOSED.value;
+        }));
+        this.canEditConfigObs = this.joinerObs.pipe(map((joiner: IJoiner): boolean => {
+            return joiner.partStatus !== PartStatus.CONFIG_PROPOSED.value;
+        }));
```

Comment:
T'es sûr que c'est la meilleurs solution de s'abonner deux fois à un même observable, c'est pas un peu sale ?

Reply:
Il faut utiliser `pipe(share())` et ça ne fait qu'un seul abonnement en vrai (sauf que là j'ai pas utilisé, donc à ajouter !)

Reply:
Rappel 📦

Reply:
J'avais rajouté le `pipe(share())` mais les observables qui en dépendent "fire" moins souvent :o Pour le moment du coup il y a plusieurs abonnements oui. Ce cas précis peut être solutionné avec un `partTypeObs = this.joinerObs.pipe(map(joiner => joiner.partType))` par exemple, mais diminue la lisibilité du code car le composant devra faire `*ngIf="(partTypeObs | async) === 0"` plutôt que `*ngIf="canReviewConfigObs | async"`. Tu préconiserais plutôt de passer par des variables modifiées dans l'abonnement donc ?

Reply:
Bah, peut être lisibility first. Sinon voir ce qui est recommandé, mais moi j'aurais été sur des variables modifiées dans l'abonnement par facilité. De toute façon la page a re-rendered a chaque modification de variable.

Reply:
C'est vers quoi je m'oriente effectivement, si j'arrive à trouver la bonne combination de tick/detectChanges/whenStable qui casse pas les tests /o\